unlock-app/src/components/interface/keychain/TransferKeyDrawer.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  AddressInput,
+  Button,
+  Drawer,
+  isAddressOrEns,
+} from '@unlock-protocol/ui'
+import { Controller, useForm } from 'react-hook-form'
+import { z } from 'zod'
+import { onResolveName } from '~/utils/resolvers'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { ethers } from 'ethers'
+import { useWalletService } from '~/utils/withWalletService'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { addressMinify } from '~/utils/strings'
+import { useMutation } from '@tanstack/react-query'
+import { useLockData } from '~/hooks/useLockData'
+
+interface TransferKeyDrawerProps {
+  isOpen: boolean
+  setIsOpen: (open: boolean) => void
+  lockAddress: string
+  network: number
+  tokenId: string
+  owner: string
+  lockName?: string
+}
+
+const TransferKeyForm = z.object({
+  newOwner: z.string({
+    description: 'Address of the new owner.',
+  }),
+})
+
+type TransferKeyFormProps = z.infer<typeof TransferKeyForm>
+
+export const TransferKeyDrawer = ({
+  isOpen,
+  setIsOpen,
+  lockAddress,
+  network,
+  tokenId,
+  owner,
+  lockName,
+}: TransferKeyDrawerProps) => {
+  const {
+    watch,
+    control,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<TransferKeyFormProps>({
+    defaultValues: {
+      newOwner: '',
+    },
+  })
+
+  const { account } = useAuth()
+  const web3Service = useWeb3Service()
+  const walletService = useWalletService()
+  const newOwner = watch('newOwner', '')
+
+  const { lock } = useLockData({
+    lockAddress,
+    network,
+  })
+
+  const maxKeysPerAddress = lock?.maxKeysPerAddress ?? 1
+
+  const onTransferFrom = async () => {
+    await walletService.transferFrom({
+      from: account!,
+      to: newOwner!,
+      lockAddress,
+      tokenId,
+    })
+  }
+
+  const transferFromMutation = useMutation(onTransferFrom)
+
+  const onTransferKeyOwnership = async () => {
+    const promise = transferFromMutation.mutateAsync()
+
+    await ToastHelper.promise(promise, {
+      loading: `Transferring key ownership to ${addressMinify(newOwner)}.`,
+      success: 'Key successfully transferred.',
+      error: 'There is an issue with transfer, please try again.',
```

Comment:
We may need to show more details about why it failed.

Reply:
Agree, for this I think we can open a different PR and handle it for all the`ToastHelper.promise`